wasmbinding/message_plugin.go

```diff
@@ -184,9 +184,6 @@ func PerformBurn(f *tokenfactorykeeper.Keeper, ctx sdk.Context, contractAddr sdk
 	if burn == nil {
 		return wasmvmtypes.InvalidRequest{Err: "burn token null mint"}
 	}
-	if burn.BurnFromAddress != "" && burn.BurnFromAddress != contractAddr.String() {
-		return wasmvmtypes.InvalidRequest{Err: "BurnFromAddress must be \"\""}
-	}
 
```

Comment:
Confirm we want to allow burn from any address if msg sender is denom admin

Reply:
Yes, except if burn from address is a module account

Reply:
creating this logic in the msg server layer